OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnPrerequisite.cs

```diff
@@ -62,6 +61,11 @@ void INotifyRemovedFromWorld.RemovedFromWorld(Actor self)
 				globalManager.Unregister(self, this, info.Prerequisites);
 		}
 
+		void INotifyOwnerChanged.OnOwnerChanged(Actor self, Player oldOwner, Player newOwner)
+		{
+			globalManager = newOwner.PlayerActor.Trait<GrantConditionOnPrerequisiteManager>();
```

Comment:
You are missing re-registration and condition revocation.
```C#
 			if (info.Prerequisites.Any())
 			{
 				globalManager.Unregister(self, this, info.Prerequisites);
 				if (conditionToken != ConditionManager.InvalidConditionToken)
 					conditionToken = conditionManager.RevokeCondition(self, conditionToken);
 			}

			globalManager = newOwner.PlayerActor.Trait<GrantConditionOnPrerequisiteManager>();
 			if (info.Prerequisites.Any())
 				globalManager.Register(self, this, info.Prerequisites);
```
Without that, this could be in the wrong state after ownership change.